src/serialized-utilities.h

```diff
@@ -0,0 +1,89 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+
+#pragma once
+#include <string>
+#include <map>
+#include <types.h>
+#include "../third-party/json.hpp"
+
+using json = nlohmann::json;
+
+namespace librealsense
+{
+	class device_interface;
+
+	namespace serialized_utilities
+	{
+		struct device_info
+		{
+			std::string name;
+			std::string product_line;
+			std::string fw_version;
+		};
+
+		class json_preset_reader
+		{
+		public:
+
+            // C'tor may throw
+			json_preset_reader( const std::string &json_content );
+
+			// Compares the device info fields on the preset json against the connected device.
+			// if not match, it throws an informative exception
+			void check_device_info( const device_interface& device ) const;
+
+			// search for a key, if found it returns an iterator to it, works together with the end() function
+			// example of use:
+			// if (reader.find(key) != reader.end())
+			//    {....}
+			json::const_iterator find(const std::string& key) const;
+			device_info get_device_info() const;
```

Comment:
Why not return as `const &`, if it's a member?

Reply:
👍